scipy/special/specfun.c

```diff
@@ -0,0 +1,8417 @@
+                }
+                *ant = c1*fx - c2*gx;
+                *bnt = sr3*(c1*fx + c2*gx);
+                if (l == 0) {
+                    *apt = *ant;
+                    *bpt = *bnt;
+                } else {
+                    *ant = -*ant;
+                    *bnt = -*bnt;
+                    x= -x;
+                }
+            }
+        } else {
+            xe = x*sqrt(x)/1.5;
+            xp6 = 1.0/sqrt(6.0*pi*xe);
+            su1 = 1.0;
+            r = 1.0;
+            xr1 = 1.0/xe;
+            for (k = 1; k < 17; k++) {
+                r = -r * xr1;
+                su1 += a[k-1]*r;
+            }
+            su2 = 1.0;
+            r = 1.0;
+            for (k = 1; k < 17; k++) {
+                r = r * xr1;
+                su2 += a[k-1]*r;
+            }
+            *apt = q0 - exp(-xe)*xp6*su1;
+            *bpt = 2.0*exp(xe)*xp6*su2;
+            su3 = 1.0;
+            r = 1.0;
+            xr2 = 1.0/(xe*xe);
+            for (k = 1; k < 9; k++) {
+                r = -r * xr2;
+                su3 += a[2*k - 1]*r;
+            }
+            su4 = a[0]*xr1;
+            r = xr1;
+            for (k = 1; k < 8; k++) {
+                r = -r * xr2;
+                su4 += a[2*k]*r;
+            }
+            su5 = su3 + su4;
+            su6 = su3 - su4;
+            *ant = q1 - q2 * xp6 * (su5*cos(xe) - su6*sin(xe));
+            *bnt = q2 * xp6 * (su5*sin(xe) + su6*cos(xe));
+        }
+    }
+    return;
+}
+
+
+void specfun_itika(double x, double *ti, double *tk) {
+
+    // =======================================================
+    // Purpose: Integrate modified Bessel functions I0(t) and
+    //          K0(t) with respect to t from 0 to x
+    // Input :  x  --- Upper limit of the integral  ( x ≥ 0 )
+    // Output:  TI --- Integration of I0(t) from 0 to x
+    //          TK --- Integration of K0(t) from 0 to x
+    // =======================================================
+
+    int k;
+    double rc1, rc2, e0, b1, b2, rs, r, tw, x2;
+    static const double a[10] = {
+        0.625, 1.0078125, 2.5927734375, 9.1868591308594,
+        4.1567974090576e+1, 2.2919635891914e+2,1.491504060477e+3,
+        1.1192354495579e+4, 9.515939374212e+4,9.0412425769041e+5
+    };
+    const double pi = 3.141592653589793;
+    const double el = 0.5772156649015329;
+
+    if (x == 0.0) {
+        *ti = 0.0;
+        *tk = 0.0;
+        return;
+    } else if (x < 20.0) {
+        x2 = x*x;
+        *ti = 1.0;
+        r = 1.0;
+        for (k = 1; k <= 50; k++) {
+            r = 0.25 * r *(2 * k - 1.0) / (2 * k + 1.0) / (k * k) * x2;
+            *ti += r;
+            if (fabs(r/(*ti)) < 1.0e-12) { break; }
+        }
+        *ti *= x;
+    } else {
+        x2 = 0.0;
+        *ti = 1.0;
+        r = 1.0;
+        for (k = 1; k <= 10; k++) {
+            r = r/x;
+            *ti += a[k-1]*r;
+        }
+        rc1 = 1.0 / sqrt(2.0*pi*x);
+        *ti = (*ti) * rc1 * exp(x);
```

Comment:
```suggestion
        *ti = (rc1 * exp(x) * (*ti);
```

https://github.com/scipy/scipy/blob/829d0dd950d63caece04526b3a617bbe2bc24023/scipy/special/specfun/specfun.f#L3780